### PR TITLE
Feature: Improve Dev-Console

### PR DESCRIPTION
**Pull request by k9ert**
I think i was a bit too fast merging the original PR. This improves on the DevConsole by:
* Most recent results now on top and not appended on bottom
* `current_user` available and also the `namespace`
* methods and properties in foldable containers

image

### CODE REVIEW

**Comment by relativisticelectron** [comment]
Executing 
image
for the first time is not compatible with `NEVER_CALLED_PYTHON_COMMAND`

image

File: src/cryptoadvance/specterext/devhelp/service.py [hunk 1]
```diff
@@ -26,6 +27,8 @@ class DevhelpService(Service):
             "app": app,
             "flask": flask,
             "flask_login": flask_login,
+            "current_user": current_user,
```
Comment (relativisticelectron) [inline src/cryptoadvance/specterext/devhelp/service.py:30]: This doesn't work:
image

It is accessible via:
image

File: src/cryptoadvance/specterext/devhelp/service.py [hunk 2]
```diff
@@ -26,6 +27,8 @@ class DevhelpService(Service):
...
+            "namespace": console.namespace,
```
Comment (relativisticelectron) [inline src/cryptoadvance/specterext/devhelp/service.py:31]: Can we remove this again?
image

**Review by relativisticelectron — COMMENTED** [review]
see comments.  For me the javascript console in any browser (developer mode) is completely sufficient (and available in any endpoint and syntax highlighting).  Is there a benefit to an in-website console?

**Comment by relativisticelectron** [comment]
Could you add the `util`  (from `import util`) to the namespace? The tools there are not accessible otherwise.

File: src/cryptoadvance/specterext/devhelp/service.py (hunk 1 above)
Comment (k9ert) [reply src/cryptoadvance/specterext/devhelp/service.py]: Fxed!

File: src/cryptoadvance/specterext/devhelp/service.py (hunk 2 above)
Comment (k9ert) [reply src/cryptoadvance/specterext/devhelp/service.py]: done

**Comment by k9ert** [comment]
> Could you add the util (from import util) to the namespace? The tools there are not accessible otherwise.

Done.

> for the first time is not compatible with NEVER_CALLED_PYTHON_COMMAND

Fixed

> For me the javascript console in any browser (developer mode) is completely sufficient (and available in any endpoint and has syntax highlighting). Is there a benefit to an in-website console?

IMHO the main 2 arguments are accessibility and convenience: 
* In a large project with many features, often enough you don't know that a feature is existing because it's badly documented (or not at all). Making it visible explicitely might be a huge deal, even if you don't use it but know that it's existing
* Even if you know that it's existing, what's the syntax again? Also, as a non frontend engineer, you might be not that familiar with the console or you might want to keep results on a different page across requests. I like the term "mechanism, not policy".

Makes sense?

**Comment by relativisticelectron** [comment]
> > For me the javascript console in any browser (developer mode) is completely sufficient (and available in any endpoint and has syntax highlighting). Is there a benefit to an in-website console?
> 
> IMHO the main 2 arguments are accessibility and convenience:
> 
>     * In a large project with many features, often enough you don't know that a feature is existing because it's badly documented (or not at all). Making it visible explicitely might be a huge deal, even if you don't use it but know that it's existing
> 
>     * Even if you know that it's existing, what's the syntax again? Also, as a non frontend engineer, you might be not that familiar with the console or you might want to keep results on a different page across requests. I like the term "mechanism, not policy".
> 
> 
> Makes sense?

Yes, makes sense :-)

**Pull request merged**